File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ""
            
        strs.sort()
        start = strs[0]
        end = strs[-1]
        ans = ""

        for i in range(min(len(start), len(end))):
            if start[i] != end[i]:
                return ans
            ans += start[i]

        return ans 

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int n = strs.size();
        int i = 0;
        string res = "";
        sort(strs.begin(),strs.end());

        while(i < min(strs[0].size(), strs[n-1].size())) {
            if(strs[0][i] == strs[n-1][i]) res+=strs[0][i];
            else break;
            i++;
        }
        return res;
    }
};